src/azureFunction/azureFunctionUtils.ts

```diff
@@ -78,6 +78,14 @@ export async function setLocalAppSetting(projectFolder: string, key: string, val
 
 export async function getAzureFunctionsExtensionApi(): Promise<AzureFunctionsExtensionApi | undefined> {
 	const apiProvider = await vscode.extensions.getExtension(constants.azureFunctionsExtensionName)?.activate() as AzureExtensionApiProvider;
+	if (!apiProvider) {
+		const response = await vscode.window.showInformationMessage(constants.azureFunctionsExtensionNotFound,
+			constants.installAzureFunction, constants.doNotInstall);
+		if (response === constants.installAzureFunction) {
+			await vscode.env.openExternal(vscode.Uri.parse(constants.linkToAzureFunctionExtension));
+		}
+		return undefined;
```

Comment:
If we install the extension you should await the command to install it and then be able to call activate again to get the API.

Although you should check to see if the AF extension requires a vs code restart. If it does then you should indicate that in the dialog and call the command to reload the window after it's done installing.